src/Database/Schema/Migrations/Filesystem.hs

```diff
@@ -85,12 +97,18 @@ migrationFromFile store name =
 -- error message.
 migrationFromPath :: FilePath -> IO (Either String Migration)
 migrationFromPath path = do
-  let name = takeBaseName $ takeFileName path
+  let name = takeBaseName path
   (Right <$> process name) `catch` (\(FilesystemStoreError s) -> return $ Left $ "Could not parse migration " ++ path ++ ":" ++ s)
 
   where
+    readMigrationFile = do
+      ymlExists <- doesFileExist (addNewMigrationExtension path)
+      if ymlExists
```

Comment:
I'm sure this could be better and `addLegacyMigrationName` is better to have extension as parameter.
Just noticed this.

Reply:
I don't know if there's something to execute a list and collect exceptions, but I'm not sure that's a good idea anyway; presumably you want to stop on the first failure, not try more things. In any case, `Control.Exception.try` might be helpful.